invenio_rdm_migrator/load/postgresql.py

```diff
@@ -0,0 +1,151 @@
+        """Cleanup csv files and DB after load."""
+        for table in self.tables:
+            table.cleanup_files(db)
+
+    def _prepare(self, entries):
+        """Dump entries in csv files for COPY command."""
+
+        _prepared_tables = []
+        for table in self._table_loads:
+            # otherwise the generator is exahusted by the first table
+            # TODO: nested generators, how expensive is this copy op?
+            _prepared_tables.extend(
+                table.prepare(self.output_dir, entries=entries)
+            )
+        
+        return iter(_prepared_tables)  # yield at the end vs yield per table
+
+    def _load(self, table_entries):
+        """Bulk load CSV table files.
+
+        Loads the tables in the order given by the generator.
+        """
+        with psycopg.connect(self.db_uri) as conn:
+            for table in table_entries:
+                name = table._table_name
+                cols = ", ".join([f.name for f in fields(table)])
+                fpath = self.output_dir / f"{name}.csv"
+                file_size = fpath.stat().st_size  # total file size for progress logging
+
+                print(f"[{_ts()}] COPY FROM {fpath}")  # TODO: logging
+                with contextlib.ExitStack() as stack:
+                    cur = stack.enter_context(conn.cursor())
+                    copy = stack.enter_context(
+                        cur.copy(f"COPY {name} ({cols}) FROM STDIN (FORMAT csv)")
+                    )
+                    fp = stack.enter_context(open(fpath, "r"))
+
+                    block_size = 8192
+
+                    def _data_blocks(block_size=8192):
+                        data = fp.read(block_size)
+                        while data:
+                            yield data
+                            data = fp.read(block_size)
+
+                    for idx, block in enumerate(_data_blocks(block_size)):
+                        if idx % 100:
+                            cur_bytes = idx * block_size
+                            percentage = (cur_bytes / file_size) * 100
+                            progress = f"{cur_bytes}/{file_size} ({percentage:.2f}%)"
+                            print(f"[{_ts()}] {name}: {progress}")
+                        copy.write(block)
+                conn.commit()
+
+    def run(self, entries, cleanup=False):
+        """Load entries."""
+        table_entries = self._prepare(entries) 
+        self._load(table_entries)
+
+        if cleanup:
+            self._cleanup()
+
+
+class DBTableLoad(ABC):
+    """Create CSV files with table create and inserts."""
+
+    def __init__(self, tables):
+        """Constructor."""
+        self._tables = tables
+
+    @abstractmethod
+    def _cleanup_db(self):
+        """Cleanup DB after load."""
+        pass
+
+    @abstractmethod
+    def _cleanup_files(self):
+        """Cleanup files after load."""
+        pass
+
+    @abstractmethod
+    def _generate_db_tuples(self, **kwargs):
```

Comment:
```suggestion
    def _generate_rows(self, **kwargs):
```

It's not tuples anymore (it's DB tuples, but rows are more common terminology).